docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,492 @@
+          - from:                                           # (6)
+              zones: ["zone-1", "zone-2"]
+            to:                                             # (7)
+              type: Only                                    # (8)
+              zones: ["zone-1", "zone-2"]
+          - from:
+              zones: ["zone-3"]
+            to:
+              type: AnyExcept
+              zones: ["zone-1"]
+          - to:
+              type: Any
+      failoverThreshold: 
+        percentage: 70                                      # (9)
+```
+
+
+(1) In `localZone` section, you configure the affinity of tags used for load balancing. If list is empty or localZone is not specified, load balancing works normally, without taking any priorities into account. 
+
+(2) In `key` field, you define the name of the tag.
+
+(3) In `weight` field, you define proportional load that the group should take. Default values are calculated based on the order of the list.
+Example:
+```yaml
+affinityTags:
+  - key: "k8s.io/node"
+  - key: k8s.io/az"
+  - key: k8s.io/region"
+```
+
+Would create weights:
+
+```yaml
+affinityTags:
+  - key: "k8s.io/node"
+    weight: 900
+  - key: "k8s.io/az"
+    weight: 90
+  - key: "k8s.io/region"
+    weight: 9
+```
+
+and all other endpoints are going to have the weight of 1. To compute the percentage of traffic that will stay inside a tag
+you need to sum all the weight within localZone and then divide weights by it. These weights don't apply for cross zone traffic, and the traffic is routed cross zone only when number of healthy instances within the local zone is lower than `failoverThreshold`. In this example sum of weights is 999. If you divide weights by it, you
+get 90% of traffic for `k8s.io/node` 9% for `k8s.io/az` and 1% for `k8s.io/region`. 
+When configuring weight you need to either specify weights for all tags or don't specify weights at all. When you specify weights the order of the list becomes irrelevant.
+
+(4) In `crossZone` section, you configure zones priority.
+
+(5) In `failover` section, you configure list of zone priorities, first rules will have the highest load balancing priority and last in order will have the lowest priority.
+
+(6) In `from` section, you configure to which zones this rule applies. This rule will apply to all DPPs in `zone-1` and `zone-2`. If not defined is for all zones.
+
+(7) In `to` section, you configure to which zones traffic should be redirected. This field is required.
+
+(8) `type` field lets you control how fallback zones are picked. Allowed values:
+- `Only` - will load balance traffic only to zones listed in `zones` field 
+- `None` - will not load balance traffic any further
+- `Any` - will load balance traffic to all zones
+- `AnyExcept` - will load balance traffic to all zones except ones specified in `zones` field
+
+(9) In `failoverThreshold` you configure the percentage of live endpoints below which we will start load balancing to the 
```

Comment:
Does this field make sense for `localZone`? Should we move this under `crossZone`?

Reply:
makes sense, changed